woocommerce/google-listings-and-ads
# Add action for Hook Documentation Generation

### Changes proposed in this Pull Request:

This PR creates an Action for generating PHP Hooks documentation when PHP files change.

It will commit a README.md in src/Hooks/README.md

### Screenshots

<img width="1478" alt="Screenshot 2023-08-17 at 15 04 53" src="https://github.com/woocommerce/google-listings-and-ads/assets/5908855/7008f9f6-3202-4886-a016-2f839fa75ae2">

<img width="1279" alt="Screenshot 2023-08-17 at 15 03 52" src="https://github.com/woocommerce/google-listings-and-ads/assets/5908855/6b47afd1-79c1-43be-bc2c-fc5ed9097606">


### Detailed test instructions:


1. Verify how the [README.md ](https://github.com/woocommerce/google-listings-and-ads/blob/0c8537f7e8d3f1b6d7ac3d651fe61d4e38b16e63/src/Hooks/README.md) has been generated in src/HOOKS/README.md 
2. Check [in the file diff ](https://github.com/woocommerce/google-listings-and-ads/pull/2060/files#diff-712190684cdb42bf9516f3e2ebe570a0fe642f133b7797dd03c31899195b3241) the links working and it's most likely as it was before. 
3. See the workflow is triggered on release branches. https://github.com/woocommerce/google-listings-and-ads/pull/2064

### Additional details:


### Changelog entry

> Dev - Add Action for Hooks Documentation Generator
